src/pages/MergePositions/Contents.tsx

```diff
@@ -264,31 +265,33 @@ export const Contents = () => {
         const conditionsOfSelectedPosition = position.conditions
 
         //#2 Filter, allow only mergeable positions
-        const positionsMergeables = positionsWithBalance.filter((p) =>
-          conditionsOfSelectedPosition.some((condition) =>
-            arePositionMergeablesByCondition(
-              [position, p],
-              condition.conditionId,
-              condition.outcomeSlotCount
+        const positionsMergeables = positionsWithBalance
+          .filter((p) =>
+            conditionsOfSelectedPosition.some((condition) =>
+              arePositionMergeablesByCondition(
+                [position, p],
+                condition.conditionId,
+                condition.outcomeSlotCount
+              )
             )
           )
-        )
+          .filter((p) => position.indexSets.length === p.indexSets.length)
 
-        const positionsFiltered = positionsMergeables.filter(
-          (positionWithBalance: PositionWithUserBalanceWithDecimals) =>
-            //#1 Filter, only positions with balance
-            !positionWithBalance.userBalanceERC1155.isZero() &&
-            //#2 Filter, remove original selected position
-            positionWithBalance.id.toLowerCase() !== position.id.toLowerCase() &&
-            //#3 Filter, positions with the same condition set
-            positionWithBalance.conditionIds.sort().join('') ===
-              position.conditionIds.sort().join('') &&
-            //#4 Filter, positions with the same collateral
-            positionWithBalance.collateralToken.toLowerCase() ===
-              position.collateralToken.toLowerCase()
-        )
+        // Calculate condition Ids to fulfill the dropdown
+        // by removing conditions if they are shared in shallower positions.
+        const indexSetsOfConditions = indexSetsByCondition(position)
+
+        const conditionIds: string[] = conditionsOfSelectedPosition
+          .map((c) => c.conditionId)
+          .filter((conditionId) => {
+            return positionsMergeables.some((p) => {
+              return indexSetsByCondition(p)[conditionId] !== indexSetsOfConditions[conditionId]
+            })
+          })
+        // Remove duplicates
+        const possibleConditions = lodashUniq(conditionIds)
 
```

Comment:
This was moved up because of the types of the positions list used to that point. Can be extracted to a function in tools but I don't know how to name it yet